cranelift/filetests/filetests/isa/x64/i128.clif

```diff
@@ -193,100 +193,92 @@ block0(v0: i128, v1: i128):
 ;   movq    %r15, 48(%rsp)
 ; block0:
 ;   cmpq    %rdx, %rdi
-;   setz    %r8b
+;   setz    %al
 ;   cmpq    %rcx, %rsi
-;   setz    %r9b
-;   andq    %r8, %r9, %r8
-;   testq   $1, %r8
+;   setz    %r8b
+;   andq    %rax, %r8, %rax
+;   testq   $1, %rax
 ;   setnz   %al
 ;   cmpq    %rdx, %rdi
-;   setnz   %r9b
+;   setnz   %r8b
 ;   cmpq    %rcx, %rsi
-;   setnz   %r10b
-;   orq     %r9, %r10, %r9
-;   testq   $1, %r9
+;   setnz   %r9b
+;   orq     %r8, %r9, %r8
+;   testq   $1, %r8
 ;   setnz   %r8b
 ;   movq    %r8, rsp(0 + virtual offset)
 ;   cmpq    %rcx, %rsi
-;   setl    %r11b
+;   setl    %r8b
 ;   setz    %r10b
 ;   cmpq    %rdx, %rdi
-;   setb    %r9b
-;   andq    %r9, %r10, %r9
-;   orq     %r9, %r11, %r9
-;   andq    %r9, $1, %r9
-;   setnz   %r9b
+;   setb    %r11b
+;   andq    %r10, %r11, %r10
+;   orq     %r8, %r10, %r8
+;   andq    %r8, $1, %r8
```

Comment:
Is the `setnz` in the original necessary here; it seems like `andq` with an immediate `$1` should be enough?

Reply:
Indeed, it seems redundant. I suspect it might have been an artifact of lowering this in separate "generate the flags" and "use in `trueif`, `brif`, ..." halves, where the `and` generates flags (and also incidentally produces the 0/1 output directly).